enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,268 @@
+   of the recommendation and its remediation steps (Insights Advisor URL
+   generated based on the cluster ID and rule ID by IO) and its value will be
+   0/1 to indicate if the recommendation is inactive/active respectively.
+5. There will be a generic/templated info-level alert present on the cluster,
+   which will present the user with brief information about the hitting
+   recommendation (its human-readable name and severity),
+   and the link for details and remediation steps.
+
+It is worth pointing out that if any key part of the Prometheus stack is not
+running as expected on the cluster, then this feature will not work. Prometheus
+manages both the metrics and the alerts, so without it, no alerts can be
+generated by IO.
+
+**Unhealthy Insights Operator scenario:**
+
+- If something goes wrong with IO, we would not want the Insights alerts to
+  remain hanging on the cluster indefinitely.
+- The alert should only fire if IO has been healthy at any point in the past 10
+  minutes. Healthy, in this context, means that there is no major error
+  happening (crashlooping, degraded operator status, etc.). A temporary error
+  state should not cause the alerts to be disabled.
+- If the user intentionally disables IO, the alerts should never fire. If IO is
+  disabled while some of the Insights alerts are firing, the alerts should
+  immediately stop firing (a part of their condition will be that IO is
+  enabled).
+- If IO was disabled by an accident, the user will be notified via the existing
+  alert.
+
+**Unhealthy Smart Proxy scenario:**
+
+- The latest successfully fetched and parsed report is too old
+  (`Meta.LastCheckedAt` is more than 8 hours ago)
+  - Reset the Insights recommendation metrics to avoid showing alerts about
+    recommendations that are no longer hitting.
+- Unable to fetch a report from Smart Proxy (endpoint unreachable, auth error, etc.)
+  - Log error, do not touch the Insights recommendation metrics.
+- The fetched report cannot be decoded into the expected structure, or the data
+  are otherwise invalid
+  - Log error, do not touch the Insights recommendation metrics.
+
+If there is anything wrong with the Smart Proxy or the reports returned by it,
+the metrics will not be modified until the validity of the last report expires.
+If IO is enabled, but something is wrong with the gathering or uploading, the
+metrics will not be directly affected for as long as the last report is valid.
+If IO has been disabled, or crashing for more then 10 minutes, the alerts will
+be disabled to avoid presenting the user with outdated information.
+
+### Risks and Mitigations
+
+- Disabling and re-enabling an alert (e.g, due to IO or Smart Proxy outage)
+  can trigger user-defined actions, even though there has not been any actual
+  change to the status of the recommendation. This could be very annoying if
+  it were to happen often or repeatedly over a short period of time.
```

Comment:
The alerting expression should be written in a way that temporary outages don't affect the alert.

Reply:
Yes, definitely, but it is very difficult to determine how long we should wait before disabling the alert as a result of an outage that is too long. If we say that 8 hours, as proposed, is enough, then repeated 9 hour outages, in between which the IO runs for an hour, let's say, would cause the alert to be repeatedly disabled and enabled. I would not want to unnecessarily over-engineer the alerts by making the expression too complex, because this is not such a big feature, and it will already be quite complicated as is.